src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -164,21 +164,30 @@ partitions_request_iterator validate_partitions(
       begin,
       valid_partitions_end,
       std::back_inserter(invalid_partitions),
-      error_code::invalid_replication_factor,
+      error_code::unknown_topic_or_partition,
       "Number of replicas does not match the topic replication factor",
       [&tp_metadata](const reassignable_partition& partition) {
           if (!partition.replicas.has_value()) {
               return true;
           }
 
-          if (
-            !tp_metadata.has_value()
-            || !tp_metadata.value().is_topic_replicable()) {
+          if (!tp_metadata.has_value()) {
+              vlog(klog.debug, "Topic metadata is undefined");
```

Comment:
please add context to your log messages. if we were trying to debug this in a production system we might, for example, want o know which topic didn't appear in the metadata.